bundles/org.openhab.binding.mqtt.homeassistant/src/main/java/org/openhab/binding/mqtt/homeassistant/internal/component/Vacuum.java

```diff
@@ -202,11 +202,11 @@ public Vacuum(ComponentFactory.ComponentConfiguration componentConfiguration) {
                 : channelConfiguration.supportedFeatures;
         List<String> deviceSupportedFeatures = Collections.emptyList();
 
-        if (!configSupportedFeatures.isEmpty()) {
+        if (configSupportedFeatures != null && !configSupportedFeatures.isEmpty()) {
```

Comment:
`configSupportedFeatures` is inferred to be `@Nullable` by Eclipse. I believe the reason is due to line 202 [`channelConfiguration.supportedFeatures`]( https://github.com/openhab/openhab-addons/pull/13518/files/d3f4d85d6ed990d5eca2d67a1428e2858600d5bd#diff-1159888df20e07eb1b066dd8a56b0214231286a00489b2f7e0b3e0fc5dcab3afR202) [being `@Nullable`](https://github.com/openhab/openhab-addons/pull/13518/files/d3f4d85d6ed990d5eca2d67a1428e2858600d5bd#diff-1159888df20e07eb1b066dd8a56b0214231286a00489b2f7e0b3e0fc5dcab3afR170)

If I would drop the null checks from line 205 and 209, I get warnings in IDE.

Reply:
Yes, eclipse assumes that value of channelConfiguration.supportedFeatures can be changed to null after null-check in line 200 and than assigned to configSupportedFeatures, because channelConfiguration.supportedFeatures is not final. 
To resolve this the value of channelConfiguration.supportedFeatures can be assigned to final local variable before line 200 and used in ternary operator.

Reply:
Sure, attached.
[13518.patch.txt](https://github.com/openhab/openhab-addons/files/9972754/13518.patch.txt)

Reply:
Appreciate that, included in commit https://github.com/openhab/openhab-addons/pull/13518/commits/77c59e5ed11f768e46b861af7f2e19288e081137

Attribution with `Co-Authored-By` commit message